### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Mhamad-Raad**
# Restaurant Menu, Development Complete 🎉🥳
## In this branch I have done the followings:
- Structured the React files.
- Implemented a design.
- Used CSS modules.
- Used the important React hooks.
- Finished the functionalities of the project.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,10 +1,27 @@
...
+
 function App() {
+  const [isShown, setIsShown] = useState(false);
+
+  const showCartHandler = () => {
+    setIsShown(true);
+  };
+
+  const hideCartHandler = () => {
+    setIsShown(false);
```
Comment (ibrahemahmad) [inline src/App.js]: try to, don't handle anything like this in App, make app.js really clear and little code.

File: src/Components/Carts/Cart.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import Modal from '../UI/Modal';
+import CartItem from './CartItem';
+
+
+const Cart = (props) => {
+
+  const cartCtx = useContext(CartContext);
+  const totalAmount = `$${cartCtx.totalAmount.toFixed(2)}`;
+  const hasItems = cartCtx.items.length > 0;
+
```
Comment (ibrahemahmad) [inline src/Components/Carts/Cart.js:19]: add null check for items, like this, item?.length
for null safety

File: src/Components/Carts/Cart.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  const removeItem = (id) => {
+    cartCtx.removeItem(id);
+  }
+
+  const addItem = (item) => {
+    console.log(item);
```
Comment (ibrahemahmad) [inline src/Components/Carts/Cart.js]: remove log, for clean code

File: src/Components/Carts/Cart.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    cartCtx.addItem({ ...item, amount: 1 });
+  }
+
+  const cartItems = <ul className={CSSClasses['cart-items']}>
```
Comment (ibrahemahmad) [inline src/Components/Carts/Cart.js]: CSSClasses.cart-items

it's also right

File: src/Components/Carts/Cart.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    {cartCtx.items.map((item) => (
+      <CartItem
```
Comment (ibrahemahmad) [inline src/Components/Carts/Cart.js:48]: items?.map
for null case

File: src/Components/Carts/Cart.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        <span>Total Amount</span>
+        <span>{ totalAmount }</span>
+      </div>
+      <div className={CSSClasses.actions}>
+        <button className={CSSClasses['button--alt']} onClick={props.onClose}>Close</button>
+        {hasItems && <button className={CSSClasses.button}>Order</button>}
+      </div>
+    </Modal>
+  );
+}
```
Comment (ibrahemahmad) [inline src/Components/Carts/Cart.js:91]: if you check this file you see, your using multiple types of using class module make it uniq

File: src/Components/Layout/HeaderCartButton.js [hunk 7]
```diff
@@ -1,15 +1,43 @@
...
+
+      return () => {
+        clearTimeout(timer);
+      };
+    }, [items]
+  );
+
+
+  const buttonClasses = `${CSSClasses.button} ${btnHighlighted===true? CSSClasses.bump: ''}`;
+
```
Comment (ibrahemahmad) [inline src/Components/Layout/HeaderCartButton.js:32]: you can using array with join, like this [CSSClasses.button,btnHighlighted? CSSClasses.bump: ''].join(' ')

File: src/Components/Meals/MealItem/MealItemForm.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+          type: 'number',
+          min: '1',
+          max: '5',
+          step: '1',
+          defaultValue: '1',
+        }}
+      />
+      <button>+ Add</button>
+      {!amountIsValid && <p>Please enter a valid amount (1-5).</p>}
+    </form>
```
Comment (ibrahemahmad) [inline src/Components/Meals/MealItem/MealItemForm.js:44]: just for getting more info you can use formik for handle any form |

File: src/index.js [hunk 9]
```diff
@@ -1,7 +1,17 @@
...
+  BrowserRouter as Router,
+  Route,
+} from 'react-router-dom';
 import './index.css';
 import App from './App';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
+  
+  <Router>
+    <Route exact path="/">
```
Comment (ibrahemahmad) [inline src/index.js]: move routes into App

**Review by ibrahemahmad — COMMENTED** [review]
check comments

**Review by darbazali — APPROVED** [review]
Although your style is a bit different from how we do it here, maybe because of you are using context API instead of a third-party state manager, but It looks great. 
Well done Kak @Mhamad-Raad

**Comment by Mhamad-Raad** [comment]
Thank you, Planning on changing the state manager to redux and currently working on it on a different project.
Thanks for the review again  @darbazali, @ibrahemahmad 😄

**Pull request merged**